### PR TITLE
[Enhancement] Simplify predicate coalesce

### PR DESCRIPTION
**Pull request by liuyehcf**
## Why I'm doing:

```sql
select * from t0 where coalesce(v1 < 2)
```

For sql like this, the function coalesce can be removed.

## What I'm doing:

Remove redundant coalesce in the predicate.

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/scalar/SimplifiedPredicateRule.java [hunk 1]
```diff
@@ -451,6 +453,34 @@ private ScalarOperator simplifiedDateTrunc(CallOperator call) {
...
+        if (ConstantOperator.FALSE.equals(notNull)) {
+            return first;
+        }
+
+        return call;
+    }
+
     // Reduce array_map whose lambda functions is trivial
     // e.g. array_map((x,y)->x, arr1,arr2) is reduced to arr1
     private static ScalarOperator arrayMap(CallOperator call) {
```
[the remark this answers is not in the document]
Comment (liuyehcf) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/scalar/SimplifiedPredicateRule.java:492]: wrong recommandation

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/PushDownPredicateScanRule.java [hunk 2]
```diff
@@ -112,4 +116,30 @@ public List<OptExpression> transform(OptExpression input, OptimizerContext conte
         OptExpression project = OptExpression.create(logicalProjectOperator, OptExpression.create(newScanOperator));
         return Lists.newArrayList(project);
     }
+
+    private ScalarOperator simplifyCoalesce(ScalarOperator predicate) {
```
Comment (Seaven) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/PushDownPredicateScanRule.java:120]: move to SimplifiedScanColumnRule?

Comment (kangkaisen) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/PushDownPredicateScanRule.java:120]: +1. we need to support simplify more exprs in the predicate.

Comment (liuyehcf) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/PushDownPredicateScanRule.java:120]: Ok

Comment (liuyehcf) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/PushDownPredicateScanRule.java:120]: this simplification cannot be performed indenpendently, for example: `where coalesce(v1, false) is NULL`, the coalesce cannot be removed directly

**Review by kangkaisen — APPROVED** [review]

**Review by packy92 — APPROVED** [review]

**Pull request merged**